$DragonFly: src/gnu/usr.bin/cc34/cc_prep/patches/loop.c.patch,v 1.4 2005/08/27 21:03:51 joerg Exp $

===================================================================
RCS file: /home/joerg/wd/repository/dragonflybsd/src/contrib/gcc-3.4/gcc/loop.c,v
retrieving revision 1.2
diff -u -r1.2 loop.c
--- loop.c	20 Dec 2004 19:23:24 -0000	1.2
+++ loop.c	20 Dec 2004 20:48:11 -0000
@@ -6525,6 +6525,14 @@
   if (GET_CODE (*mult_val) == USE)
     *mult_val = XEXP (*mult_val, 0);
 
+#ifndef FRAME_GROWS_DOWNWARD
+  if (flag_propolice_protection
+      && GET_CODE (*add_val) == PLUS
+      && (XEXP (*add_val, 0) == frame_pointer_rtx
+	  || XEXP (*add_val, 1) == frame_pointer_rtx))
+    return 0;
+#endif
+
   if (is_addr)
     *pbenefit += address_cost (orig_x, addr_mode) - reg_address_cost;
   else
